components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/DAO/GroupDAO.java

```diff
@@ -589,6 +589,42 @@ public String[] getGroupNameList(String searchAttributeName, String searchAttrib
         return roleList.toArray(new String[0]);
     }
 
+    /**
+     * List the groups created from SCIM with the group name filter.
+     *
+     * @param tenantId        Tenant ID.
+     * @param groupNameFilter Group name filter to be applied.
+     * @return List of SCIM group names.
+     * @throws IdentitySCIMException If an error occurred while reading the SCIM Group information.
+     */
+    public String[] getGroupNameList(Integer tenantId, String groupNameFilter)
+            throws IdentitySCIMException {
+
+        List<String> roleList = new ArrayList<>();
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection()) {
+            try (PreparedStatement prepStmt = connection.prepareStatement(
+                    SQLQueries.LIST_SCIM_GROUPS_SQL_BY_ROLE_NAME)) {
+                prepStmt.setInt(1, tenantId);
+                prepStmt.setString(2, groupNameFilter);
+
+                try (ResultSet rSet = prepStmt.executeQuery()) {
+                    while (rSet.next()) {
+                        String roleName = rSet.getString(1);
+                        if (StringUtils.isNotEmpty(roleName)) {
+                            // Remove the primary domain name from roleNames.
+                            roleList.add(removePrimaryDomainName(roleName));
+                        }
+                    }
+                }
+            }
+        } catch (SQLException e) {
+            log.error("Error when executing the SQL : " + SQLQueries.LIST_SCIM_GROUPS_SQL_BY_ROLE_NAME);
```

Comment:
```suggestion
```
This error log may not required. The required information has been passed to the IdentitySCIMException

Reply:
Ack

Reply:
Addressed by https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/619/commits/e3c6ccc6e808e5e5b382476f35ef988bccbd42f1